x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -503,19 +485,14 @@ public void invalidateAccessToken(String tokenString, ActionListener<TokensInval
         } else {
             maybeStartTokenRemover();
             final Iterator<TimeValue> backoff = DEFAULT_BACKOFF.iterator();
-            try {
-                decodeToken(tokenString, ActionListener.wrap(userToken -> {
-                    if (userToken == null) {
-                        listener.onFailure(traceLog("invalidate token", tokenString, malformedTokenException()));
-                    } else {
-                        indexInvalidation(Collections.singleton(userToken.getId()), listener, backoff,
-                            "access_token", null);
-                    }
-                }, listener::onFailure));
-            } catch (IOException e) {
-                logger.error("received a malformed token as part of a invalidation request", e);
-                listener.onFailure(malformedTokenException());
-            }
+            decodeToken(tokenString, ActionListener.wrap(userToken -> {
+                if (userToken == null) {
+                    logger.error("received a malformed token as part of a invalidation request");
```

Comment:
I do not like that we log two messages here. One at error and one at trace.

Reply:
I have removed the error message. There are other places where the same could be said, but I find the format of trace entries as uninformative checkpoint typeish, that you probably need when something very boggling is happening. They seem to be scattered in the failure handlers and try to achieve consistency, but sometimes the warn, or error messages offer much more information.

So I kept the others as they were.